src/layouts/Modal.tsx

```diff
@@ -0,0 +1,140 @@
+import { useTranslation } from "react-i18next";
+
+type Props = {
+  children?: React.ReactNode;
+  // closeable?: boolean;
+  closeFunc: () => void;
+  backFunc?: () => void;
+  submitFunc?: () => void;
+  title: string;
+  submitLabel?: string;
+};
+
+const Modal: FC<Props> = ({
+  // closeable = true,
+  closeFunc,
+  backFunc,
+  children,
+  title,
+  submitLabel,
+  submitFunc,
+}) => {
+  const { t } = useTranslation();
+
+  const cancelButtonRef = useRef(null);
+
+  return (
+    <Transition.Root show as={Fragment}>
+      <Dialog
+        as="div"
+        className="relative z-10"
+        initialFocus={cancelButtonRef}
+        onClose={() => closeFunc()}
+      >
+        <Transition.Child
+          as={Fragment}
+          enter="ease-out duration-300"
+          enterFrom="opacity-0"
+          enterTo="opacity-100"
+          leave="ease-in duration-200"
+          leaveFrom="opacity-100"
+          leaveTo="opacity-0"
+        >
+          <div className="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity" />
+        </Transition.Child>
+
+        <div className="fixed inset-0 z-10 overflow-y-auto">
+          <div className="flex min-h-full items-end justify-center p-4 text-center sm:items-center sm:p-0">
+            <Transition.Child
+              as={Fragment}
+              enter="ease-out duration-300"
+              enterFrom="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+              enterTo="opacity-100 translate-y-0 sm:scale-100"
+              leave="ease-in duration-200"
+              leaveFrom="opacity-100 translate-y-0 sm:scale-100"
+              leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+            >
+              <Dialog.Panel className="relative transform  overflow-hidden rounded-lg bg-white text-left shadow-xl transition-all sm:my-8 sm:w-full sm:max-w-lg">
+                <form
+                  onSubmit={(event) => {
+                    event.preventDefault();
+                    submitFunc && submitFunc();
+                  }}
+                >
+                  <div className="absolute right-0 top-0 hidden pr-4 pt-4 sm:block">
+                    <button
+                      type="button"
+                      className="rounded-md bg-white dark:bg-zinc-900 text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
+                      onClick={() => closeFunc()}
+                    >
+                      <span className="sr-only">{t("navigation.cancel")}</span>
+                      <XMarkIcon className="h-6 w-6" aria-hidden="true" />
+                    </button>
+                  </div>
+
+                  <div className="bg-white px-4 pb-4 pt-5 dark:bg-gray-800 dark:text-white sm:p-6 sm:pb-4">
+                    <div className="sm:flex sm:items-start">
+                      <div className="mt-3 text-center sm:ml-4 sm:mt-0 sm:text-left">
+                        <Dialog.Title
```

Comment:
Maybe the title should be bigger & centered in desktop as well (same with txbuttons).

Reply:
- bigger is good
- I'm not so sure about "centered on desktop"
  Looking at [some known examples](https://component.gallery/components/modal/) I don't find a centered exmaple for desktop. If in the future bigger modals are used centered text will float weirdly. On smaller screens (i.e. mobile) this isn't an issue.

Reply:
Sure I'm fine with it :)